### ISSUE TITLE
Problem notification integrations return error in v1.36.0

### ISSUE THREAD

**Issue by mattBaumBeneva** [dynatrace-oss/terraform-provider-dynatrace#307]
**Describe the bug**
Since migrating to provider version 1.36.0, our build returns errors for the OpsGenie and xMatters notification ressources.

The Ops Genie resources returns:
<img width="655" alt="image" src="https://github.com/dynatrace-oss/terraform-provider-dynatrace/assets/127420012/26b3e5b0-7359-4520-bb9b-3f240b87cd32">

The xMatters resource only indicates something is null, without specifying which field.

We have printed out the Alerting Profile ID, and it is not null.

Here is a code sample:
```
resource "dynatrace_ops_genie_notification" "team" {
  count  = var.opsgenie_notif != "" ? 1 : 0
  name   = "Opsgenie - ${var.team} - ${var.nom_alertage} - [piloté par terraform]"
  active = true
  lifecycle {
    ignore_changes = [
      active
    ]
  }
  api_key = var.opsgenie_notif
  domain  = "api.opsgenie.com"
  message = "{ProblemImpact} Problem {ProblemID} ({ProblemSeverity}): {ProblemTitle}"
  profile = dynatrace_alerting.team.id
}
```

**Expected behavior**
The integration should be configured as before.

**Comment by mattBaumBeneva** [comment]
I can confirm that rolling back to 1.35.3 avoids the issue.

**Comment by Dynatrace-Reinhard-Pilz** [comment]
Hello @mattBaumBeneva,

You're right. We introduced behavior with 1.36.0 that instructs the REST APIs to behave more lenient in case attributes are missing or are unexpected.

A side effect we didn't notice was, that this also disables other functionalities - in this case translating an alerting profile ID in the format of Settings 2.0 into its legacy ID. Until we can get that fixed by Dynatrace R&D on the server side we've decided to roll back that feature.

Release v1.36.1 is about to get released which will take care of your issue. Apologies for the inconvenience.

best regards,
Reinhard
